File: frontend/src/types/support.ts
export type SupportTicketStatus = 'open' | 'pending' | 'closed'
export type SupportTicketPriority = 'low' | 'normal' | 'high'

export interface SupportTicketMessage {
  id: string
  ticket_id: string
  author_user_id?: string | null
  author_role: 'customer' | 'staff' | string
  body: string
  created_at?: string | null
}

export interface SupportTicket {
  id: string
  customer_id: string
  environment_id?: string | null
  subject: string
  status: SupportTicketStatus | string
  priority: SupportTicketPriority | string
  created_at?: string | null
  updated_at?: string | null
  customer_email?: string | null
  customer_name?: string | null
  messages?: SupportTicketMessage[]
}

File: backend/app/schemas/support.py
"""Support ticket schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import Field

from app.schemas.common import SchemaBase


class SupportTicketCreateRequest(SchemaBase):
    subject: str = Field(min_length=3, max_length=255)
    body: str = Field(min_length=3, max_length=20000)
    priority: str = Field(default="normal", max_length=16)
    department: str | None = Field(default=None, max_length=64)
    environment_id: UUID | None = None


class SupportTicketMessageCreateRequest(SchemaBase):
    body: str = Field(min_length=1, max_length=20000)
    send_direct_message: bool = False


class SupportTicketMessageResponse(SchemaBase):
    id: UUID
    ticket_id: UUID
    author_user_id: UUID | None = None
    author_role: str
    body: str
    created_at: datetime | None = None


class SupportTicketResponse(SchemaBase):
    id: UUID
    customer_id: UUID
    environment_id: UUID | None = None
    subject: str
    status: str
    priority: str
    created_at: datetime | None = None
    updated_at: datetime | None = None
    customer_email: str | None = None
    customer_name: str | None = None
    messages: list[SupportTicketMessageResponse] = Field(default_factory=list)
